packages/invoices/src/OutboundShipment/DetailView/modals/hooks/useDraftOutboundLines.tsx

```diff
@@ -3,81 +3,90 @@ import {
   generateUUID,
   InvoiceLineNodeType,
   Item,
-  StockLine,
   useConfirmOnLeaving,
   useDirtyCheck,
-  useParams,
 } from '@openmsupply-client/common';
 import { useStockLines } from '@openmsupply-client/system';
-import { InvoiceLine, DraftOutboundLine } from '../../../../types';
+import { DraftOutboundLine } from '../../../../types';
 import { sortByExpiry, issueStock } from '../utils';
-import { useOutboundLines } from '../../../api';
+import { useOutboundLines, useOutboundFields } from '../../../api';
+import {
+  PartialOutboundLineFragment,
+  PartialStockLineFragment,
+} from '../../../api/operations.generated';
 
 export const createPlaceholderRow = (
   invoiceId: string,
-  itemId = '',
   id = generateUUID()
 ): DraftOutboundLine => ({
   __typename: 'InvoiceLineNode',
-  availableNumberOfPacks: 0,
-  batch: 'Placeholder',
-  costPricePerPack: 0,
+  batch: '',
   id,
-  itemId,
-  onHold: false,
   packSize: 1,
   sellPricePerPack: 0,
-  storeId: '',
-  totalNumberOfPacks: 0,
   numberOfPacks: 0,
   isCreated: true,
   isUpdated: false,
-  stockLineId: 'placeholder',
   invoiceId,
-  itemCode: '',
-  itemName: '',
   type: InvoiceLineNodeType.UnallocatedStock,
+  item: { id: '', code: '', name: '', __typename: 'ItemNode' },
 });
 
 interface DraftOutboundLineSeeds {
   invoiceId: string;
-  stockLine?: Partial<StockLine> | null;
-  invoiceLine?: Partial<InvoiceLine> | null;
+  invoiceLine: PartialOutboundLineFragment;
 }
 
-export const createDraftOutboundLine = ({
+export const createDraftOutboundLineFromStockLine = ({
```

Comment:
I attempted to simplify these functions.

- When creating a draft line, it's either from a stock line (when there is no existing invoice line), or an invoice line
- In the case of a stock line, we need to create an invoice line, which is relatively straight forward
- In the case of an existing invoice line, we just need to clone it (with the exception of available number of packs)

So I split this into two functions and it feels far easier to understand

Reply:
yeah - I like it. much clearer